### PR TITLE
fix(findImage) : Fix findImage bug introduced in #2960

### PR DESCRIPTION
**Pull request by aravindmd**
- fix bug introduced in #2960  , add the `clouddriver` check after synthetic stages are computed

### CODE REVIEW

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/tasks/DeploymentDetailsAware.groovy [hunk 1]
```diff
@@ -42,7 +42,7 @@ trait DeploymentDetailsAware {
     Closure callback) {
     Stage previousStage = getPreviousStageWithImage(stage, targetRegion, targetCloudProvider)
     def result = [:]
-    if (previousStage) {
+    if (previousStage && isCloudProviderEqual(stage, previousStage)) {
```
Comment (jeyrschabu) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/tasks/DeploymentDetailsAware.groovy]: Might be able to get away with just ` previousStage.context.cloudProvider == stage.context.cloudProvider` here and get rid of the function.

Comment (aravindmd) [reply orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/tasks/DeploymentDetailsAware.groovy]: i had to throw in the function to add those `null` checks , some tests pass `null` for `cloudprovider`

Comment (marchello2000) [reply orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/tasks/DeploymentDetailsAware.groovy]: what happens if:
```
Stage A          ->       Stage B                -> Stage C
findimage(aws)        findImage(titus)        deploy(aws)
```

won't stage C find image from stage B only to discard it because the cloud providers don't match?
In other words if the getPreviousStageWithImage returned the wrong image and we have to continue searching this change won't work? (I am probably missing something)

Comment (aravindmd) [reply orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/tasks/DeploymentDetailsAware.groovy]: this case works fine , i can discuss this with you offline.
